packages/api-request/src/middlewares/preloading.js

```diff
@@ -36,9 +36,10 @@ const createPreloadingMiddleware = ( preloadedData ) => ( options, next ) => {
 		const method = options.method || 'GET';
 		const path = getStablePath( options.path );
 
-		if ( 'GET' === method && preloadedData[ path ] ) {
+		if ( preloadedData[ method ] && preloadedData[ method ][ path ] ) {
 			const deferred = jQuery.Deferred();
-			deferred.resolve( preloadedData[ path ].body );
+			const response = 'OPTIONS' === method ? preloadedData[ method ][ path ] : preloadedData[ method ][ path ].body;
```

Comment:
We should have a unit test for this. I'm also not sure why we ought to need to special-case OPTIONS requests. Who's to say we don't want other headers from the request? Just seems like we're making inconsistency for not an entirely valid reason.